Android Source/mobile/src/main/java/de/jo2k/loggingmodule/DataHandler.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Basic implementation of a simple logger that writes a JSON stream to the mobile's external storage.
+ * JSON might be non-trivial to parse, but the data structure that can be generated per sensor
+ * are not uniform so it is probably best to stay flexible.
+ *
+ * Contributed by Jonathan Liebers (c) 2021, git@jo2k.de
+ *
+ * Thanks to `Steppschuh` (Stephan Schultz) for the great initial work.
+ *
+ * License: Apache License 2.0
+ */
+
+package de.jo2k.loggingmodule;
+
+import android.os.Environment;
+import android.util.Log;
+
+import net.steppschuh.datalogger.data.DataBatch;
+import net.steppschuh.sensordatalogger.BuildConfig;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import static net.steppschuh.sensordatalogger.PhoneApp.TAG;
+
+public class DataHandler {
+    private boolean isRecording = false;
+    private FileWriter filewriter;
+    private String filename;
+    private boolean firstEntryWasWritten;
+
+    public void logDataBatch(DataBatch dataBatch, String sourceNodeId) {
+        /**
```

Comment:
This style of block comment (used for method documentation) needs to be placed right above the method, not in the method body.

Reply:
Got it, wrote to much Python in the past apparently where it is flipped around. :+1: